DependencyInjection/SmartboxIntegrationFrameworkExtension.php

```diff
@@ -229,45 +234,43 @@ protected function loadQueueDrivers(ContainerBuilder $container)
-                        throw new \InvalidArgumentException('You need to define at least one connection to use the AMQP driver.');
+                        throw new \InvalidArgumentException('You need to define at least one connection to use the PHP AMQP lib driver.');
                     }
 
-                    $queueManager = $container->findDefinition('smartesb.amqp.queue_manager');
+                    $driverDef = new Definition(PhpAmqpLibDriver::class);
+
                     foreach ($driverConfig['connections'] as $index => $uri) {
-                        $connection = parse_url($uri);
-                        $connectionId = "$driverId.connection.$index";
-
-                        $container->register($connectionId, 'AMQPConnection')->setArguments([[
-                            'host' => $connection['host'],
-                            'port' => $connection['port'] ?? 5672,
-                            'vhost' => trim($connection['path'] ?? '', '/'),
-                            'login' => $connection['user'],
-                            'password' => $connection['pass'],
-                        ]]);
-
-                        $queueManager->addMethodCall('addConnection', [new Reference($connectionId)]);
+                        $driverDef->addMethodCall('setPort', [$driverConfig['port'] ?? AsyncQueueDriverInterface::DEFAULT_PORT]);
+                        $driverDef->addMethodCall('configure', [
+                            $driverConfig['host'],
+                            $driverConfig['username'],
+                            $driverConfig['password'],
+                            $driverConfig['vhost']
+                        ]);
                     }
-                    $this->useAmqp = true;
 
-                    $driverDef = $container->findDefinition('smartesb.drivers.queue.amqp');
-                    $driverDef->addMethodCall('setId', [$driverId]);
-                    $driverDef->addMethodCall('configure', [null, null, null, $driverConfig['format']]);
+                    $queueDriverRegistry->addMethodCall('setDriver', [$driverName, $driverDef]);
 
-                    $queueDriverRegistry->addMethodCall('setDriver', [$driverName, new Reference($driverId)]);
+                    $driverDef->addMethodCall('setId', [$driverId]);
+                    $driverDef->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
+                    $driverDef->addMethodCall('setFormat', [$driverConfig['format']]);
 
                     $container->setDefinition($driverId, $driverDef);
-                    $container->findDefinition('smartesb.protocols.queue')
-                        ->addMethodCall('setDefaultConsumer', [new Reference('smartesb.consumers.async_queue')]);
+
+                    $consumer = $container->findDefinition('smartesb.async_consumers.queue');
+                    $consumer->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
```

Comment:
The serializer is already defined in the consumer service definition (Resources/config/consumers.yml)

Reply:
Done.